### ISSUE TITLE
Calculation of mean orientations in spatial.transform.Rotation

### ISSUE THREAD

**Issue by pmla** [scipy/scipy#10862]
For most vector data calculation of a mean is trivial - we just use `np.mean`.  Calculation of a mean for orientational/rotational data is just as useful, but conceptually more complicated.

The problem (and solutions) are discussed in the excellent review article by Hartley et al.
https://users.cecs.anu.edu.au/~hartley/Papers/PDF/Hartley-Trumpf:Rotation-averaging:IJCV.pdf
Sections 4 and 5 contain the relevant material.

There are many different averaging functions which can be used.  The most common choices are the geodesic L2, quaternion L2, and chordal L2 means.  Each comes with its own set of pros and cons.  Here is a brief summary of the properties that I consider the most relevant:

```
         mean  | intrinsic mean?  unique solution?              closed-form solution?
===============|==========================================================================
  geodesic L2  | yes              in ball of radius pi / 2      no - requires optimization
quaternion L2  | no               in ball of radius pi / 2      yes
   chordal L2  | no               yes (barring special cases)   yes
```
(in ball of radius pi / 2) means that the mean is unique if the orientations are not spread too far apart.

I propose that we implement the chordal L2 mean.  It has O(n) running time and is simple to implement.  If we decide to add other means later, we can add a keyword to choose which one to calculate.

**Comment by pmla** [comment]
@nmayorov pinging you in here.

**Comment by pmla** [comment]
Closing since we merged #10866

**Comment by isarandi** [comment]
There seems to be a contradiction between the docs and the implementation. The code seems to average the quaternions, which, according to my understanding, is the quaternion L2 variant, while the docs and the above plan talk about chordal L2.
